zap/src/irgen/des.rs

```diff
@@ -18,16 +22,57 @@ impl Gen for Des<'_> {
 	where
 		I: Iterator<Item = &'a Ty<'src>>,
 	{
+		self.new_scope();
+
 		for (ty, name) in types.zip(names) {
 			self.push_ty(ty, Var::Name(name.to_string()));
 		}
 
-		self.buf
+		self.end_scope();
+
+		self.buf.output()
 	}
 
 	fn get_var_occurrences(&mut self) -> &mut HashMap<String, usize> {
 		self.var_occurrences
 	}
+
+	fn new_scope(&mut self) {
+		let scope_buf = OutputBuffer::new();
+		self.buf.push(scope_buf.clone());
```

Comment:
~~Why are you pushing here? There's nothing to add to the buffer as it's an empty `Vec`.~~
Does pushing here make sense? It wasn't apparant why you did that at a first glance, but there's no real alternative right?